lib/osutil/atomic.go

```diff
@@ -25,26 +25,36 @@ var (
 // returned on Close, so a lazy user can ignore errors until Close.
 type AtomicWriter struct {
```

Comment:
Not necessarily part of this PR but we should probably consider renaming this as there is no guarantee whatsoever that this operation is "atomic" after involving the `Filesystem`. Maybe it should become a WithTempFileWriter or some crap in the future.